test/e2e/install_test.go

```diff
@@ -99,10 +109,137 @@ var _ = Describe("Operator Install", func() {
 
 		})
 		AfterEach(func() {
+			if basePath := env.GetString("ARTIFACT_PATH", ""); basePath != "" {
+				// get all the artifacts from the test run and save them to the artifact path
+				getArtifactsOutput(basePath)
+			}
 			err := c.Delete(ctx, operatorCatalog)
 			Expect(err).ToNot(HaveOccurred())
 			err = c.Delete(ctx, operator)
 			Expect(err).ToNot(HaveOccurred())
 		})
 	})
 })
+
+// getArtifactsOutput gets all the artifacts from the test run and saves them to the artifact path.
+// right now it will save:
+// - operators
+// - pods logs
+// - deployments
+// - bundle
+// - bundledeployments
+// - catalogsources
+
+func getArtifactsOutput(basePath string) {
+	ctx := context.Background()
+	cfg = ctrl.GetConfigOrDie()
+	kubeClient, err := kubeclient.NewForConfig(cfg)
+	Expect(err).To(Not(HaveOccurred()))
+
+	artifactPath := path.Join(basePath, fmt.Sprint(time.Now().UnixNano())) + "/"
+
+	// Create the full artifact path
+	err = os.MkdirAll(artifactPath, 0755)
+	Expect(err).To(Not(HaveOccurred()))
+
+	// Get all namespaces
+	namespaces, err := kubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
```

Comment:
nit: you limit use of kubeClient only for getting pod logs, if you wanted everything else to be consistent

Reply:
fixed